### PR TITLE
Fix attachment uri in trx if same attachment filename is same

### PR DESCRIPTION
**Pull request by smadala**
- Update the attachment filename with targetFilename while moving all the attachments to one directory for trx report.

Fixes: https://github.com/Microsoft/vstest/issues/1626

### CODE REVIEW

File: src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/Converter.cs [hunk 1]
```diff
@@ -148,6 +148,11 @@ public static TrxObjectModel.TestOutcome ToOutcome(ObjectModel.TestOutcome rockS
 
         public static List<CollectorDataEntry> ToCollectionEntries(IEnumerable<ObjectModel.AttachmentSet> attachmentSets, TestRun testRun, string trxFileDirectory)
         {
+            if (EqtTrace.IsInfoEnabled)
+            {
+                EqtTrace.Info($"Converter.ToCollectionEntries: Converting attachmentSets {string.Join(",", attachmentSets)} to collection entries.");
```
Comment (nigurr) [inline src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/Converter.cs]: attachmentSets null check is below and here we are printing before null check?

File: src/Microsoft.TestPlatform.ObjectModel/AttachmentSet.cs [hunk 2]
```diff
@@ -40,6 +40,11 @@ public AttachmentSet(Uri uri, string displayName)
             DisplayName = displayName;
             Attachments = new List<UriDataAttachment>();
         }
+
+        public override string ToString()
+        {
+            return $"{nameof(Uri)}: {Uri.AbsoluteUri}, {nameof(DisplayName)}: {DisplayName}, {nameof(Attachments)}: [{ string.Join(",", Attachments)}]";
```
Comment (nigurr) [inline src/Microsoft.TestPlatform.ObjectModel/AttachmentSet.cs]: can Uri be null?

File: src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/Converter.cs (hunk 1 above)
Comment (smadala) [reply src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/Converter.cs]: Thanks. added null check.

File: src/Microsoft.TestPlatform.ObjectModel/AttachmentSet.cs (hunk 2 above)
Comment (smadala) [reply src/Microsoft.TestPlatform.ObjectModel/AttachmentSet.cs]: Not sure about that. However we are using AbsoluteUri in [other places](https://github.com/smadala/vstest/blob/2cfc8a2abc01adb740bd9f89c7cbcffd69e4eaac/src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/Converter.cs#L164) without any check. Not adding null check here.

**Review by nigurr — APPROVED** [review]

**Comment by smadala** [comment]
@dotnet-bot test this please.

**Comment by smadala** [comment]
@dotnet-bot test this please

**Pull request merged**